solidity/contracts/PhasedEscrow.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+
+interface IBeneficiaryContract {
+    function __escrowSentTokens(uint256 amount) external;
+}
+
+// @title PhasedEscrow
+// @notice A token holder contract allowing contract owner to set beneficiary of
+//         tokens held by the contract and allowing the owner to withdraw the
+//         tokens to that beneficiary in phases.
+contract PhasedEscrow is Ownable {
+    using SafeERC20 for IERC20;
+
+    event BeneficiaryUpdated(address beneficiary);
+    event TokensWithdrawn(address beneficiary, uint256 amount);
+
+    IERC20 public token;
+    IBeneficiaryContract public beneficiary;
+
+    constructor(IERC20 _token) public {
+        token = _token;
+    }
+
+    // @notice Sets the provided address as a beneficiary allowing it to
+    //         withdraw all tokens from escrow. This function can be called only
+    //         by escrow owner.
+    function setBeneficiary(IBeneficiaryContract _beneficiary) external onlyOwner {
+        beneficiary = _beneficiary;
+        emit BeneficiaryUpdated(address(beneficiary));
+    }
+
+    // @notice Withdraws the specified number of tokens from escrow to the
+    //         beneficiary. If the beneficiary is not set, or there are
+    //         insufficient tokens in escrow, the function fails.
+    function withdraw(uint256 amount) external onlyOwner {
+        require(address(beneficiary) != address(0), "Beneficiary not assigned");
+
+        uint256 balance = token.balanceOf(address(this));
+        require(amount <= balance, "Not enough tokens for withdrawal");
+
+        token.safeTransfer(address(beneficiary), amount);
+        emit TokensWithdrawn(address(beneficiary), amount);
+
+        beneficiary.__escrowSentTokens(amount);
```

Comment:
`Rewards` contract expects that tokens are received via `approveAndCall + receiveApproval`.

We can just `safeTransfer` tokens there but then we'll have to call `receiveApproval` from some other account.

I wonder if a better solution wouldn't be to have an alternative to `withdraw` that would do `approveAndCall`.

Reply:
This is for a liquidation rewards contract that hasn't been written yet, so I think we can set it up accordingly. The goal here was basically maximum simplicity so that we can start the rewards flow. That said, we're talking about prioritizing the staking rewards adjustments over this, so it may make sense to pause this anyway.